src/tests/functional/inference_engine/snippets/src/pass/lowering.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <common_test_utils/ngraph_test_utils.hpp>
+#include "pass/lowering.hpp"
+#include "snippets/pass/collapse_subgraph.hpp"
+
+
+namespace ngraph {
+namespace builder {
+namespace subgraph {
+
+DummyTargetMachine::DummyTargetMachine() {
+    auto dummy_functor = [this](const std::shared_ptr<ngraph::Node>& n) {
+        return std::make_shared<DummyEmitter>();
+    };
+    jitters[op::v0::Parameter::get_type_info_static()] = dummy_functor;
+    jitters[op::v0::Constant::get_type_info_static()] = dummy_functor;
+    jitters[op::v0::Result::get_type_info_static()] = dummy_functor;
+    jitters[op::v1::Add::get_type_info_static()] = dummy_functor;
+    jitters[op::v1::Subtract::get_type_info_static()] = dummy_functor;
+    jitters[op::v1::Multiply::get_type_info_static()] = dummy_functor;
+    jitters[op::v1::Multiply::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::Load::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::VectorLoad::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::ScalarLoad::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::BroadcastLoad::get_type_info_static()] = dummy_functor;
+
+    jitters[ngraph::snippets::op::Store::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::VectorStore::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::ScalarStore::get_type_info_static()] = dummy_functor;
+
+    jitters[ngraph::snippets::op::Scalar::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::BroadcastMove::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::Kernel::get_type_info_static()] = dummy_functor;
+    jitters[ngraph::snippets::op::Tile::get_type_info_static()] = dummy_functor;
+}
+
+void SnippetsLoweringTests::prepare() {
+        ASSERT_TRUE(function);
+        // Check that the function is fully tokenizable and obtain subgraph
+        tokenize(function);
+        getSubgraph(function);
+}
+Shape SnippetsLoweringTests::canonicalize(BlockedShapeVector& input_blocked_shapes, BlockedShapeVector& output_blocked_shapes) {
+        return subgraph->canonicalize(output_blocked_shapes, input_blocked_shapes);
+}
+void SnippetsLoweringTests::lower() {
+        subgraph->set_generator(std::make_shared<DummyGenerator>());
+        subgraph->generate();
+        function = subgraph->get_body();
+}
+
+void SnippetsLoweringTests::tokenize(std::shared_ptr<Model>& f) {
+        ngraph::pass::Manager m;
+        m.register_pass<ngraph::snippets::pass::EnumerateNodes>();
+        m.register_pass<ngraph::snippets::pass::TokenizeSnippets>();
+        m.run_passes(f);
+}
+
+void SnippetsLoweringTests::getSubgraph(std::shared_ptr<Model>& f) {
```

Comment:
Here I wanted to avoid confusion with initOriginal() (also initReference and InitLowered) from SnippetsFunctionBase, because this function doesn't really initialize subgraph, it just finds already existing subgraph in the function.
If you think that this name is misleading I don't mind renaming, lets discuss an alternative offline.

Reply:
Converted to static member as discussed in https://github.com/openvinotoolkit/openvino/pull/10605#discussion_r825850184